apps/wallet/src/ui/app/components/network-selector/index.tsx

```diff
@@ -1,75 +1,132 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { useFeature } from '@growthbook/growthbook-react';
 import cl from 'classnames';
-import { useMemo, useCallback } from 'react';
+import { motion, AnimatePresence } from 'framer-motion';
+import { useMemo, useCallback, useState } from 'react';
 
+import { CustomRPCInput } from './custom-rpc-input';
 import { API_ENV_TO_INFO, API_ENV } from '_app/ApiProvider';
-import Icon from '_components/icon';
+import { FEATURES } from '_app/experimentation/features';
+import Icon, { SuiIcons } from '_components/icon';
 import { useAppSelector, useAppDispatch } from '_hooks';
 import { changeRPCNetwork } from '_redux/slices/app';
 
 import st from './NetworkSelector.module.scss';
 
+const EXCLUDE_STAGING = process.env.SHOW_STAGING !== 'false' && API_ENV.staging;
+
+type NetworkTypes = keyof typeof API_ENV;
+
 const NetworkSelector = () => {
-    const selectedApiEnv = useAppSelector(({ app }) => app.apiEnv);
-    const dispatch = useAppDispatch();
-    const netWorks = useMemo(
-        () =>
-            Object.keys(API_ENV)
-                .filter(
-                    (env) =>
-                        process.env.SHOW_STAGING !== 'false' ||
-                        env !== API_ENV.staging
-                )
-                .map((itm) => ({
-                    style: {
-                        color: API_ENV_TO_INFO[itm as keyof typeof API_ENV]
-                            .color,
-                    },
-                    ...API_ENV_TO_INFO[itm as keyof typeof API_ENV],
-                    networkName: itm,
-                })),
-        []
+    const [selectedApiEnv, customRPC] = useAppSelector(({ app }) => [
+        app.apiEnv,
+        app.customRPC,
+    ]);
+    const [showCustomRPCInput, setShowCustomRPCInput] = useState<boolean>(
+        selectedApiEnv === API_ENV.customRPC
     );
 
+    const [activeClass, setActiveClass] = useState<string>(selectedApiEnv);
+
+    const dispatch = useAppDispatch();
+    const excludeCustomRPC =
```

Comment:
Just make this a boolean instead of evaluated into an `API_ENV` value. This approach is more confusing than it's worth.

Reply:
Oh, I should change the label `activeClass`, which refers to the selected/active network name, which may differ from the saved network name.